llvm/lib/Target/AMDGPU/GCNSchedStrategy.cpp

```diff
@@ -1089,33 +1102,224 @@ bool ClusteredLowOccStage::initGCNSchedStage() {
 #define REMAT_PREFIX "[PreRARemat] "
 #define REMAT_DEBUG(X) LLVM_DEBUG(dbgs() << REMAT_PREFIX; X;)
 
+#if !defined(NDEBUG) || defined(LLVM_ENABLE_DUMP)
+void PreRARematStage::printTargetRegions(bool PrintAll) const {
+  if (PrintAll) {
+    for (auto [I, Target] : enumerate(RPTargets))
+      REMAT_DEBUG(dbgs() << "  [" << I << "] " << Target << '\n');
```

Comment:
You only use this under REMAT_DEBUG uses, so can drop all the REMAT_DEBUGs here

Reply:
Then wrap those dangling uses in the _DEBUG blocks, they don't need their own ifndef NDEBUGs